librosa/core/constantq.py

```diff
@@ -971,11 +979,11 @@ def vqt(
             window=window,
             gamma=gamma,
             dtype=dtype,
-            alpha=alpha
+            alpha=alpha,
         )
 
         # Re-scale the filters to compensate for downsampling
-        fft_basis[:] *= np.sqrt(2 ** (i-oct_start))
+        fft_basis[:] *= np.sqrt(sr / my_sr)
```

Comment:
this change is transparent iff `hop_length` is a power of two but fixes the #1381 bug when it isn't— do i read this right?

Reply:
Yeah exactly. We had accidentally been carrying the assumption that each octave is at a different sampling rate and using the implicitly calculated ratio to compensate. Here we just use the explicit ratio of rates, which will always do the right thing.